src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -282,7 +284,12 @@ public static TestResultSummary parseAndSummarize(@Nonnull JUnitTask task, Pipel
             }
 
             if (!task.isSkipPublishingChecks()) {
-                new JUnitChecksPublisher(build, task.getChecksName(), result, summary).publishChecks(listener);
+                // If we haven't been provided with a checks name, and we have pipeline test details, set the checks name
+                // to be a ' / '-joined string of the enclosing blocks names. JUnitChecksPublisher will handle defaults if
+                // checksName ends up being empty or null
+                String checksName = task.getChecksName() != null || pipelineTestDetails == null ? task.getChecksName() :
```

Comment:
example PR where this has been tested?

Reply:
I'm going to spend some quality time with mockito to see if I can get the checks tests testing more of the implementation of `parseAndSummarize`.